src/tests/fs_tests.rs

```diff
@@ -4036,4 +4036,322 @@ pub mod fs_tests {
+
+        let cage = interface::cagetable_getref(1);
+
+        // Create a directory and try to seek within it.
+        let path = "/test_dir";
+        assert_eq!(cage.mkdir_syscall(path, S_IRWXA), 0);
+        let fd = cage.open_syscall(path, O_RDONLY, S_IRWXA);
+        assert!(fd >= 0);
+
+        // Attempt to seek within the directory and check if it succeeds
+        assert_eq!(cage.lseek_syscall(fd, 1, SEEK_SET), 1);
+
+        assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        lindrustfinalize();
+    }
+
+    #[test]
+    pub fn ut_lind_fs_lseek_invalid_whence() {
+        // acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+
+        // Test to create a file and check for invalid `whence` value
+        let fd = cage.open_syscall("/test_file", O_CREAT | O_RDWR, S_IRWXA);
+        assert!(fd >= 0);
+
+        // Attempt to seek with an invalid `whence` value and check if it returns an
+        // error
+        assert_eq!(
+            cage.lseek_syscall(fd, 10, 999), // Invalid whence value
+            -(Errno::EINVAL as i32)
+        );
+
+        assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        lindrustfinalize();
+    }
+
+    #[test]
+    pub fn ut_lind_fs_lseek_beyond_file_size() {
+        // acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+
+        // Test to create a file and seek beyond its size
+        let fd = cage.open_syscall("/test_file", O_CREAT | O_RDWR, S_IRWXA);
+        assert!(fd >= 0);
+
+        // Write sample data to the file.
+        assert_eq!(cage.write_syscall(fd, str2cbuf("hello"), 5), 5);
+
+        // Seek beyond the end of the file and verify if it succeeds
+        assert_eq!(
+            cage.lseek_syscall(fd, 10, SEEK_END),
+            15 // 5 (file size) + 10 (offset)
+        );
+
+        assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        lindrustfinalize();
+    }
+
+    #[test]
+    pub fn ut_lind_fs_lseek_before_start_of_file() {
+        // acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+
+        // Test to create a file and attempt to seek before the start of the file
+        let fd = cage.open_syscall("/test_file", O_CREAT | O_RDWR, S_IRWXA);
+        assert!(fd >= 0);
+
+        // Attempt to seek to a negative offset and check if it returns an error
+        assert_eq!(
+            cage.lseek_syscall(fd, -10, SEEK_SET),
+            -(Errno::EINVAL as i32)
+        );
```

Comment:
adding a test case that uses seek_curr with a negative offset that falls before 0 would also be good I guess?

Reply:
Added